libs/datasets/ca_vaccination_backfill.py

```diff
@@ -0,0 +1,85 @@
+import dataclasses
+import pandas as pd
+from libs.datasets.timeseries import MultiRegionDataset
+from libs.datasets import AggregationLevel
+from libs.pipeline import Region
+
+from covidactnow.datapublic.common_fields import CommonFields
+from covidactnow.datapublic.common_fields import FieldName
+from covidactnow.datapublic.common_fields import PdFields
+
+
+def derive_ca_county_vaccine_pct(ds_in: MultiRegionDataset) -> MultiRegionDataset:
+    """Derives vaccination metrics for CA counties based on State 1st vs 2nd dose reporting."""
+
+    ca_county_dataset = ds_in.get_subset(aggregation_level=AggregationLevel.COUNTY, state="CA")
+    ca_county_wide = ca_county_dataset.timeseries_wide_dates()
+    fields_to_check = [
+        CommonFields.VACCINATIONS_INITIATED,
+        CommonFields.VACCINATIONS_COMPLETED,
+        CommonFields.VACCINATIONS_INITIATED_PCT,
+        CommonFields.VACCINATIONS_COMPLETED_PCT,
+    ]
+    # Assert that possible fields we want to estimate are all NA - if one of these is
+    # not NA, likely do not need to estimate anymore and this methodology can be removed.
+    assert ca_county_wide.loc[(slice(None), fields_to_check), :].isna().all().all()
+
+    ca_state_wide = ds_in.get_regions_subset([Region.from_state("CA")]).timeseries_wide_dates()
+    fields = [
+        CommonFields.VACCINATIONS_INITIATED,
+        CommonFields.VACCINATIONS_COMPLETED,
+        CommonFields.VACCINES_ADMINISTERED,
+    ]
+    ca_data = ca_state_wide.loc[(slice(None), fields), :]
```

Comment:
ca_state_wide isn't used elsewhere. the code below works with county and state data so it'd be nice to use the name ca_state_wide instead of ca_data below.
Try using `reset_index(CommonFields.LOCATION_ID, drop=True)` instead of droplevel so you can use one less local name for state data, then stop using the name ca_data.